integration-libs/cdc/root/service/cdc-js.service.ts

```diff
@@ -489,17 +500,11 @@ export class CdcJsService implements OnDestroy {
    * Obtain the email of the currently logged in user
    * @returns emailID of the loggedIn user
    */
-  protected getLoggedInUserEmail() {
-    let email: string = '';
-    this.userProfileFacade
-      .get()
-      .pipe(take(1))
-      .subscribe((user?: User) => {
-        if (user && user.uid) {
-          email = user.uid;
-        }
-      });
-    return email;
+  protected getLoggedInUserEmail(): Observable<User> {
+    return this.userProfileFacade.get().pipe(
+      take(1),
+      filter((user): user is User => Boolean(user))
```

Comment:
```suggestion
      filter((user): user is User => Boolean(user))
      take(1),
```

:et's swap those 2 operators. Otherwise, if the first emission is empty and the second real user, we won't return the second emission. It's because the operator `take(1)` would close the stream after first emission